functions/replace-route/tests/test_replace_route.py

```diff
@@ -215,3 +236,247 @@ def test_disable_ipv6():
+    nft_table_start = "nft_nat_table=table ip nat {\nchain postrouting {\ntype nat hook postrouting priority 100; policy accept;"
+    masquerade_rule = "\nip saddr 10.0.0.0/8 oifname \"eth0\" counter packets 0 bytes 0 masquerade"
+    nft_table_end = "\n}\n}\n"
+
+    # Build diagnostic outputs
+    good_nft_output = nft_table_start + masquerade_rule + nft_table_end
+    missing_masquerade_output = nft_table_start + nft_table_end
+
+    # Test scenarios: (description, ssm_output, source_dest_check, expected_result)
+    test_cases = [
+        ("successful diagnostics", f"ip_forward=1\n{good_nft_output}", False, True),
+        ("ip_forward=0 failure", f"ip_forward=0\n{good_nft_output}", False, False),
+        ("missing masquerade rule failure", f"ip_forward=1\n{missing_masquerade_output}", False, False),
+        ("source/dest check enabled failure", f"ip_forward=1\n{good_nft_output}", True, False),
+        ("error in source/dest check", f"ip_forward=1\n{good_nft_output}", None, False),
+    ]
+
+    with mock.patch('boto3.client') as mock_boto_client:
+        mock_ssm = mock.MagicMock()
+        mock_boto_client.return_value = mock_ssm
+        mock_ssm.send_command.return_value = {'Command': {'CommandId': 'test-command-id'}}
+
+        for description, ssm_output, source_dest_check, expected_result in test_cases:
+            mock_ssm.get_command_invocation.return_value = {
+                'Status': 'Success',
+                'StandardOutputContent': ssm_output,
+                'StandardErrorContent': ''
+            }
+
+            with mock.patch('app.is_source_dest_check_enabled', return_value=source_dest_check):
+                result = run_nat_instance_diagnostics('i-12345678')
+                assert result == expected_result, f"Failed test case: {description}"
+
+        # Test SSM command failure separately
+        mock_ssm.send_command.side_effect = botocore.exceptions.ClientError(
+            {'Error': {'Code': 'InvalidInstanceId', 'Message': 'Test error'}},
+            'SendCommand'
+        )
+        result = run_nat_instance_diagnostics('i-12345678')
+        assert result == False
+
+
+@mock_aws
+@mock.patch('time.sleep')
+def test_attempt_nat_instance_restore(mock_sleep):
+    from app import attempt_nat_instance_restore
+
+    # Need to mock boto3.client to avoid calling AWS API
+    with mock.patch('boto3.client') as mock_boto_client:
+        # Mock AWS clients
+        mock_ssm = mock.MagicMock()
+        mock_ec2 = mock.MagicMock()
+
+        def get_boto_client(service):
+            if service == 'ssm':
+                return mock_ssm
+            elif service == 'ec2':
+                return mock_ec2
+            return mock.MagicMock()
+
+        mock_boto_client.side_effect = get_boto_client
+
+    # Setup environment
+    route_tables = ['rtb-12345', 'rtb-67890']
+    os.environ["ROUTE_TABLE_IDS_CSV"] = ",".join(route_tables)
+    os.environ["NAT_ASG_NAME"] = "test-nat-asg"
```

Comment:
wdyt about using [pytest's monkeypatch](https://docs.pytest.org/en/stable/how-to/monkeypatch.html#monkeypatching-environment-variables) for this?

Reply:
Good suggestion! Done in [e66c3fc](https://github.com/chime/terraform-aws-alternat/pull/139/commits/e66c3fc6b706dbd0d5c6daca55f733b1e03c4986)